src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -965,12 +965,12 @@ class AnnotationCollector : public ResourceObj{
   // Set the annotation name:
   void set_annotation(ID id) {
     assert((int)id >= 0 && (int)id < (int)_annotation_LIMIT, "oob");
-    _annotations_present |= nth_bit((int)id);
+    _annotations_present |= (int)nth_bit((int)id);
   }
 
   void remove_annotation(ID id) {
     assert((int)id >= 0 && (int)id < (int)_annotation_LIMIT, "oob");
-    _annotations_present &= ~nth_bit((int)id);
+    _annotations_present &= (int)~nth_bit((int)id);
```

Comment:
This silences the warning but without a runtime check.  It would be nice to have a constexpr template version of nth_bit that return the desire type and also do a range check on the input, unlike this macro version always returns intx and returns 0 if the shift is out of range.

Reply:
Let's fine a separate RFE for nth_bit and fix all of these at once.  This appears a lot.